pkg/modules/modules.go

```diff
@@ -0,0 +1,155 @@
+package modules
+
+import (
+	"context"
+	"errors"
+
+	"github.com/grafana/dskit/modules"
+	"github.com/grafana/dskit/services"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// List of available targets.
+const (
+	All string = "all"
+)
+
+type Engine interface {
```

Comment:
@wbrowne i ended up pulling in `Engine` and `Manager` from one of your PRs to fix a test failure